File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int n = s.length();
        int j=0;
        int maxi=0;
        HashSet<Character> set = new HashSet<>();
        for(int i=0; i<n; i++){
            while(set.contains(s.charAt(i))){
                set.remove(s.charAt(j));
                j++;
            }
            set.add(s.charAt(i));
           
                maxi=Math.max(maxi, i-j+1);
            
        }
        return maxi;
    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        seen = set()
        i=0
        j=0
        count=0
        maxi=0
        for i, ch in enumerate(s):
            while ch in seen:
                seen.remove(s[j])
                j+=1
            seen.add(ch)
            
            maxi = max(maxi, i-j+1)
        return maxi
            
        